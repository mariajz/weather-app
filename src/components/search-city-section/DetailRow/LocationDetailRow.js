import React from 'react';
import { CustomIcon } from '../../../commons/visual-elements';
import { Divider } from '../../../commons/styles';
import {
    Row,
    StyledText,
    LocationUnavailableRow,
} from './LocationDetailRow.style';

const LocationDetailRow = ({
    handleOnDropDownItemPress,
    location,
    locationAvailable,
}) => {
    return (
        <If condition={locationAvailable}>
            <>
                <Row
                    testID="row"
                    onPress={() => {
                        handleOnDropDownItemPress(location);
                    }}>
                    <CustomIcon iconName="pin" size={12} tintColor="black" />
                    <StyledText>
                        {location?.name},{location?.country}
                    </StyledText>
                </Row>
                <Divider />
            </>
            <Else />
            <LocationUnavailableRow testID="unavailable-row">
                <CustomIcon iconName="unavailable" size={16} tintColor="red" />
                <StyledText>
                    Weather details unavailable for entered location. Please try
                    another closer location.
                </StyledText>
            </LocationUnavailableRow>
        </If>
    );
};

export default LocationDetailRow;
