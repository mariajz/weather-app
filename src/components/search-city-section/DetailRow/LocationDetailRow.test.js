import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import LocationDetailRow from './LocationDetailRow';

jest.mock('../../../commons/visual-elements', () => ({
    CustomIcon: props => <MockView {...props}>Search Icon</MockView>,
}));

jest.mock('../../../commons/styles', () => ({
    Divider: () => <MockView>Divider</MockView>,
}));

const mockHandleOnDropDownItemPress = jest.fn();
const defaultProps = {
    location: { country: 'India', key: 5, name: 'Delhi' },
    locationAvailable: true,
    handleOnDropDownItemPress: mockHandleOnDropDownItemPress,
};

describe('LocationDetailRow', () => {
    it('should render LocationDetailRow when locationAvailable is true', () => {
        const container = render(<LocationDetailRow {...defaultProps} />);
        const { getByTestId } = container;

        expect(container).toMatchSnapshot();

        fireEvent.press(getByTestId('row'));

        expect(mockHandleOnDropDownItemPress).toHaveBeenCalledTimes(1);
    });

    it('should render LocationUnavailableRow when locationAvailable is false', () => {
        const modifiedProps = {
            locationAvailable: false,
        };
        const container = render(<LocationDetailRow {...modifiedProps} />);
        const { getByTestId } = container;

        expect(getByTestId('unavailable-row')).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
