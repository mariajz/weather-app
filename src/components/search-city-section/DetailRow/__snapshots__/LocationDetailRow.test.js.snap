// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetailRow should render LocationDetailRow when locationAvailable is true 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 25,
        "marginLeft": 10,
        "opacity": 1,
      }
    }
    testID="row"
  >
    <View
      iconName="pin"
      size={12}
      tintColor="black"
    >
      Search Icon
    </View>
    <Text
      style={
        {
          "marginLeft": 5,
        }
      }
    >
      Delhi
      ,
      India
    </Text>
  </View>,
  <View>
    Divider
  </View>,
]
`;

exports[`LocationDetailRow should render LocationUnavailableRow when locationAvailable is false 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "marginLeft": 10,
      "width": 300,
    }
  }
  testID="unavailable-row"
>
  <View
    iconName="unavailable"
    size={16}
    tintColor="red"
  >
    Search Icon
  </View>
  <Text
    style={
      {
        "marginLeft": 5,
      }
    }
  >
    Weather details unavailable for entered location. Please try another closer location.
  </Text>
</View>
`;
